src/styles/style.css

```diff
@@ -0,0 +1,269 @@
+* {
+  box-sizing: border-box;
+}
+
+body {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+  margin: 0;
+  padding: 0;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+  max-width: 707px;
+}
+
+.calendar__day {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  background-color: #eee;
+  height: 100px;
+  width: 100px;
+  border: 1px solid black;
+}
+
+.calendar__day:hover {
+  cursor: pointer;
+  background-color: #ffbfcb;
+  transform: translateY(-20px);
+  animation-name: transform;
+  animation-duration: 500ms;
+}
+
+.calendar--month-length-27 .calendar__day:nth-child(n + 28) {
+  display: none;
+}
+
+.calendar--month-length-28 .calendar__day:nth-child(n + 29) {
+  display: none;
+}
+
+.calendar--month-length-29 .calendar__day:nth-child(n + 30) {
+  display: none;
+}
+
+.calendar--month-length-30 .calendar__day:nth-child(n + 31) {
+  display: none;
+}
+
+.calendar--start-day-mon .calendar__day:first-child {
+  margin-left: 0;
+}
+
+.calendar--start-day-tue .calendar__day:first-child {
+  margin-left: 101px;
+}
+
+.calendar--start-day-wed .calendar__day:first-child {
+  margin-left: 202px;
+}
+
+.calendar--start-day-thu .calendar__day:first-child {
+  margin-left: 303px;
+}
+
+.calendar--start-day-fri .calendar__day:first-child {
+  margin-left: 404px;
+}
+
+.calendar--start-day-sat .calendar__day:first-child {
+  margin-left: 505px;
+}
+
+.calendar--start-day-sun .calendar__day:first-child {
+  margin-left: 606px;
+}
+
+.calendar__day:nth-child(1)::before {
+  content: '1';
+  font-family: Arial, Helvetica, sans-serif;
```

Comment:
Repeated styles for the `font-family` and `font-size` in `.calendar__day:nth-child(n)::before` can be combined into a single CSS rule. This will make your code DRY (Don't Repeat Yourself), which is a best practice in coding.